### ISSUE TITLE
full_reference_name/short_reference_name ignore stropping

### ISSUE THREAD

**Issue by pavel-kirienko** [UAVCAN/nunavut#75]
Create a DSDL namespace named `str`. Suppose a template contains this:

```python
{% for t, _ in T.get_nested_types() -%}
from {{ t|full_reference_name }} import {{ t|short_reference_name }} as {{ t|short_reference_name }}
{% endfor -%}
```

The generated code will be malformed because `full_reference_name` (and its short counterpart) yield `str` whereas the generated namespace is named `str_`.

**Comment by thirtytwobits** [comment]
You'd need to use the id filter, right?

```
{% for t, _ in T.get_nested_types() -%}
from {{ t|full_reference_name |id }} import {{ t|short_reference_name|id }} as {{ t|short_reference_name|id }}
{% endfor -%}
```

**Comment by pavel-kirienko** [comment]
The dot operator gets in the way:

```python
from uavcanZX002EdiagnosticZX002ERecord_1_0 import Record_1_0 as Record_1_0
```
...supposed to be:

```python
from uavcan.diagnostic.Record_1_0 import Record_1_0 as Record_1_0
```

**Comment by thirtytwobits** [comment]
ah

**Comment by thirtytwobits** [comment]
I think this should work if you set your `--target-language` then you can use the id filter on the namespace properly. I just tested this locally and it works (preserves dots).

**Comment by pavel-kirienko** [comment]
I think I am already doing (the equivalent of) that here:

https://github.com/UAVCAN/pyuavcan/blob/04efe8c7b85c63806a7c2a44e4c84b89340e5273/pyuavcan/dsdl/_compiler.py#L174-L187

In PyUAVCAN, Nunavut is used as a library.

**Comment by pavel-kirienko** [comment]
Fixed in #86.
